### PR TITLE
[Kernel] Triton MLA logits workspace

### PR DESCRIPTION
**Pull request by NickLucche**
`TritonMLAImpl.forward_mqa` allocates a fresh float32 scratch tensor attn_logits on every decode forward call.
This PR addresses the TODO https://github.com/vllm-project/vllm/blob/bf292b5f6b537d154fc09a3b232f89cbc66827f5/vllm/v1/attention/backends/mla/triton_mla.py#L185-L196

by allocating a logits workspace ahead of time, using a single buffer that layer can share. This makes it so that we don't pay the allocation price at every step. 
The approach is the ~same as in turboquant/rocm, in which we allocate a max-num-seqs*dims buffer (worst case traffic as usual).

I've measured a ~35%/1.35x boost, which seems pretty big imo so happy to re-run a bigger sweep if we need more data points.

  @LucasWilkinson @MatthewBonanni 

## Test With
Unit
```
pytest tests/v1/attention/test_mla_backends.py -k triton
```
Eval
```
lm_eval --model local-completions --tasks gsm8k \
      --model_args model=deepseek-ai/DeepSeek-V2-Lite-Chat,base_url=http://127.0.0.1:8000/v1/completions,num_concurrent=100,max_retries=3,tokenized_requests=False,tokenizer_backend=None

#( matches main in test_accuracy.py)
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.6755|±  |0.0129|
|     |       |strict-match    |     5|exact_match|↑  |0.6672|±  |0.0130|
```

Perf deepseek-ai/DeepSeek-V2-Lite-Chat on B200, 1 GPU
```
vllm serve deepseek-ai/DeepSeek-V2-Lite-Chat --attention-backend TRITON_MLA
vllm bench serve --random-input-len 1024 --random-output-len 1024

# pre
============ Serving Benchmark Result ============
Successful requests:                     1000
Failed requests:                         0
Benchmark duration (s):                  80.31
Total input tokens:                      1024000
Total generated tokens:                  1024000
Request throughput (req/s):              12.45
Output token throughput (tok/s):         12750.89
Peak output token throughput (tok/s):    22488.00
Peak concurrent requests:                1000.00
Total token throughput (tok/s):          25501.78
---------------Time to First Token----------------
Mean TTFT (ms):                          25999.56
Median TTFT (ms):                        26016.09
P99 TTFT (ms):                           30883.07
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          49.26
Median TPOT (ms):                        49.20
P99 TPOT (ms):                           49.92
---------------Inter-token Latency----------------
Mean ITL (ms):                           49.26
Median ITL (ms):                         46.32
P99 ITL (ms):                            117.48
==================================================
# post
============ Serving Benchmark Result ============
Successful requests:                     1000
Failed requests:                         0
Benchmark duration (s):                  59.37
Total input tokens:                      1024000
Total generated tokens:                  1024000
Request throughput (req/s):              16.84
Output token throughput (tok/s):         17246.78
Peak output token throughput (tok/s):    23558.00
Peak concurrent requests:                1000.00
Total token throughput (tok/s):          34493.55
---------------Time to First Token----------------
Mean TTFT (ms):                          4992.24
Median TTFT (ms):                        4935.52
P99 TTFT (ms):                           9771.75
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          49.36
Median TPOT (ms):                        49.43
P99 TPOT (ms):                           50.04
---------------Inter-token Latency----------------
Mean ITL (ms):                           49.36
Median ITL (ms):                         46.61
P99 ITL (ms):                            117.02
==================================================
```

### CODE REVIEW

**Review by LucasWilkinson — APPROVED** [review]
LGTM , thanks!

**Comment by NickLucche** [comment]
Failures are unrelated
```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================= 1 passed, 21 warnings in 204.84s (0:03:24) ==============================================
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
(worktree2) root@slurm-eus-04a-prod-b200-192-207:~/upstream/worktree2# pytest -v -s tests/v1/tracing
```
